### PR TITLE
Implemented get_partitioning_additional_data:#138

### LINKED ISSUE #138: Create DB function to gather partitioning additional data
## Background
We need to read the partitioning information.

## Feature
Create DB function to gather partitioning additional data

## Proposed Solution [Optional]
Function signature
```
CREATE OR REPLACE FUNCTION runs.create_partitioning_if_not_exists(
    IN  i_partitioning          JSONB,
    OUT ad_name                 TEXT,
    OUT ad_value                TEXT
) RETURNS SET OF record AS

```

### PR DESCRIPTION
**Pull request by TebaleloS**
* Implemented `get_partitioning_additional_data` DB function
* Added test cases using balta library
closes #138

### CODE REVIEW

**Review by salamonpavel — COMMENTED** [review]
Hi @TebaleloS, just a friendly nudge about the crucial role of testing before we submit a pull request. It's like our safety net, catching any bugs and making the review process smoother. Remember, we've got the Balta library on our side for database testing. I also noticed the code in this PR seems to be a work in progress, which makes testing even more vital and honestly, there's no reason to present unfinished code for review. 

If there's ever a moment of uncertainty about your task, don't hesitate to ask your teammates. We're all in this together and ready to help. But also, let's not forget that we each own our tasks. So, let's ensure we're putting our best foot forward to maintain our high-quality codebase.

File: database/src/main/postgres/runs/V1.5.12_get_partitioning_additional_data.sql [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        status_text := 'The partitioning does not exist.';
+        RETURN;
+    END IF;
+
+    status = 11;
+    status_text = 'OK';
+
+    RETURN QUERY
+        SELECT mdh.ad_name, mdh.ad_value, status, status_text
+        FROM runs.measure_definitions_history AS mdh
```
Comment (lsulak) [inline database/src/main/postgres/runs/V1.5.12_get_partitioning_additional_data.sql:57]: I don't think such table exists

Comment (TebaleloS) [reply database/src/main/postgres/runs/V1.5.12_get_partitioning_additional_data.sql:57]: Yah, it was just a skeleton code. I was going to enquire about that.

File: database/src/main/postgres/runs/V1.5.12_get_partitioning_additional_data.sql [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+-- Function: runs.get_partitioning_additional_data(2)
+--      Iterates over a JSONB object and returns each key-value pair as a record
+--
+-- Parameters:
+--      i_partitioning      - JSONB object where each key is an additional data name and its corresponding value is the additional data value
+--
+-- Returns:
+--      ad_name             - Name of the additional data
+--      ad_value            - Value of the additional data
+--
```
Comment (benedeki) [inline database/src/main/postgres/runs/V1.5.12_get_partitioning_additional_data.sql:36]: Missing status descriptions

File: database/src/main/postgres/runs/V1.5.12_get_partitioning_additional_data.sql [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    _fk_partitioning                    BIGINT;
+BEGIN
+    _fk_partitioning = runs._get_id_partitioning(i_partitioning);
+
+    IF _fk_partitioning IS NULL THEN
+        ad_name := 'Partitioning not found';
+        ad_value := '';
+        status := 41;
+        status_text := 'The partitioning does not exist.';
+        RETURN;
```
Comment (benedeki) [inline database/src/main/postgres/runs/V1.5.12_get_partitioning_additional_data.sql:49]: Same as in #137
```suggestion
        status := 41;
        status_text := 'The partitioning does not exist.';
        RETURN NEXT;
        RETURN;
```

File: database/src/main/postgres/runs/V1.5.12_get_partitioning_additional_data.sql (hunk 1 above)
Comment (benedeki) [reply database/src/main/postgres/runs/V1.5.12_get_partitioning_additional_data.sql:57]: Yeah wrong query.
If you create an unfinished PR, please mark it as `Work in progress` (I already added it here)

File: database/src/test/scala/za/co/absa/atum/database/runs/GetPartitioningAditionalDataTest.scala [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+  test("Get partitioning additional data returns additional data for partitioning with additional data") {
+    table("runs.partitionings").insert(
+      add("partitioning", partitioning1)
+        .add("created_by", "Joseph")
+    )
+
+    val fkPartitioning: Long = table("runs.partitionings").fieldValue("partitioning", partitioning1, "id_partitioning").get.get
+
+    table("runs.additional_data").insert(
```
Comment (benedeki) [inline database/src/test/scala/za/co/absa/atum/database/runs/GetPartitioningAditionalDataTest.scala:63]: I would add another record for the partitioning (to return at least 2).

And just to be extra diligent what about creating a record for a different `fkPartitioning` (you can fake it, you don't have to create another partitioning). Just to be sure that one is NOT returned.

**Review by benedeki — COMMENTED** [review]
Good job.
Code is good. Would just enhance one of the test cases.

**Review by benedeki — APPROVED** [review]
Looks good!

**Pull request merged**